### PR TITLE
Added APIM Subscription resource

### PR DESCRIPTION
**Pull request by NickGraham101**

### CODE REVIEW

File: azure/template.json [hunk 1]
```diff
@@ -62,6 +68,30 @@
...
+                },
+                "parameters": {
+                    "apimName": {
+                        "value": "[parameters('sharedApimName')]"
+                    },
+                    "subscriptionName": {
+                        "value": "[variables('funcAppName')]"
+                    },
+                    "subscriptionScope": {
+                        "value": "[concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', parameters('sharedApimResourceGroup'), '/providers/Microsoft.ApiManagement/service/', 'das-at-shared-apim', '/products/EmployerIncentivesOuterApi')]"
```
Comment (jack-education) [inline azure/template.json:90]:
```suggestion
                        "value": "[concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', parameters('sharedApimResourceGroup'), '/providers/Microsoft.ApiManagement/service/', parameters('sharedApimName'), '/products/EmployerIncentivesOuterApi')]"
```

File: azure/template.json [hunk 2]
```diff
@@ -32,6 +32,12 @@
         "serviceName": {
             "type": "string"
         },
+        "sharedApimName": {
```
Comment (jack-education) [inline azure/template.json:35]:
```suggestion
        "sharedApimName": {
```
Can this match directly with the variable name in variable group so no extra variable is needed in pipeline variables

Comment (NickGraham101) [reply azure/template.json:35]: Why do they need to match?  In das-employer-incentives-web I've done -sharedApimName "$(SharedApimServiceName)"

Comment (jack-education) [reply azure/template.json:35]: If we can reduce the number of pipeline variables we should.

I think das-employer-incentives-web should change as well.

We follow that pattern for other variable group variables, eg. SharedManagementResourceGroup/ / SharedEnvResourceGroup

Comment (NickGraham101) [reply azure/template.json]: I'ved altered the param name

**Pull request merged**